### ISSUE TITLE
Some HMTL mails are not loaded

### ISSUE THREAD

**Issue by etekinbas** [Foundry376/Mailspring#569]
Last week, I said to myself "I finally found what i'm looking for" but it looks like MailSpring have some huge bugs on rendering HTML mails. 

Bug 1: Some HTML mails are completely invisible. I mean you can't even see the mails. I've checked their HTML code and also i go further and send the same HTML mail to Default Mac Mail, Gmail, Yahoo, Hotmail and Yandex. They all render it and show the mail perfectly. But MailSpring hiding all the content when i open it.

![screen shot 2018-01-25 at 10 13 32](https://user-images.githubusercontent.com/226711/35375531-6cb1efd0-01b8-11e8-87d0-f02fb25a3b63.png)

Bug 2: Most of the HTML signature images are de-attached from the mail and shown as attachment. This is the first i see something like this. It's a very strange bug.

![screen shot 2018-01-25 at 10 12 31](https://user-images.githubusercontent.com/226711/35375517-5e02fccc-01b8-11e8-98b6-28bfa394409a.png)

##### Are there any related issues?

...

##### What operating system are you using?
MacOs - High Sierra

##### What version of Mailspring are you using?
Version 1.1.0

--

**Bug?**
##### Do you have any third-party plugins installed? If so, which ones?
No

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
Don't know. I'm using Gmail

##### Is the issue reproducible with a particular attachment, message, signature, etc?

![screen shot 2018-01-25 at 10 13 32](https://user-images.githubusercontent.com/226711/35375590-a2d47fd8-01b8-11e8-8739-f78775f804e3.png)
![screen shot 2018-01-25 at 10 12 31](https://user-images.githubusercontent.com/226711/35375591-a2f7835c-01b8-11e8-94a5-4243ebc376bc.png)

--

**Feature Request?**
##### Does this feature exist in another mail client or tool you use?
Yes

**Comment by umutbesler** [comment]
Exactly same problem here. Some of the HTML mails are not shown.

**Comment by jerm** [comment]
Another example of this is emails from tripit.com:

http://share.jermops.com/1K1q2T433A0a (blurry bits are my doing)

**Comment by bengotow** [comment]
Hmm strange indeed - I'll see if I can reproduce this locally.

**Comment by jacoboAR** [comment]
Same issue for me

**Comment by bengotow** [comment]
Hey folks - I've been able to reproduce this and the next version will include a fix. Essentially the images are being referenced from the message body in an unusual way that Mailspring didn't recognize, so it threw them at the bottom instead.

**Comment by jerm** [comment]
Oh interesting.. I guess mine was different, as there's nothing at the bottom, it's just blank gray line of sadness: http://share.jermops.com/1K1q2T433A0a

**Comment by bengotow** [comment]
Hmm that is interesting... I take that back, it looks like my solution so far only corrects issue #2 above. I'll see if there's something separate going on for that first scenario where the entire email is gone.

**Comment by umutbesler** [comment]
Hello Ben, if you need i can send you an example mail that Mailspring views empty.

**Comment by bengotow** [comment]
Hey! @umutbesler that'd actually be really helpful - I haven't been able to reproduce that part of this ticket. Could you send me one to ben@foundry376.com? If you have a mail client like Thunderbird or Apple Mail that supports "redirecting" messages, that'd be ideal since it preserves the message exactly. (Gonna try to add that feature to Mailspring soon.)

**Comment by umutbesler** [comment]
Hello @bengotow, i've both send the message directly and redirect from Apple mail.

**Comment by jaap** [comment]
I'm experiencing the same sort of behaviour on html mails.
The message doesn't get the right height and a lot of content is hidden from view.
![cloudapp annotation 2018-02-15 at 1 20 17 pm png](https://user-images.githubusercontent.com/724492/36256481-7bc19aec-1253-11e8-8dbf-17a867b8258b.png)

**Comment by BenjaminGK** [comment]
Any plans of when this will be released? It has been pending release for 23 days now.

**Comment by Monastario** [comment]
I have exactly the same thing as "jaap"a little bit can bee seen, but the rest of the message is not visible. Last comment is from Feb 9. Is someone doing something about this? Cause the client is unusable if you cannot open emails.

**Comment by joho** [comment]
I'm also experiencing the exact same issue with tripit emails. In general I've noticed lots of render variations between mailspring and other clients I use (gmail, inbox, evolution on fedora, airmail on macOS) of varying annoyance levels.

**Comment by bengotow** [comment]
Hey folks—I took another pass at fixing this earlier this week and I believe version 1.3 (which is shipping in the next day or so) will resolve this for good. I was able to reproduce the clipping in @jaap's screenshot reliably with a handful of emails, and it turns out that Chromium is firing the "loaded" event a bit earlier than it used to, and when Mailspring asks for the height of the email frame, the reported number doesn't include the images that were just loaded.

Sorry for the hassle, will post here again when the fix is available.
